test/go.mod

```diff
@@ -16,6 +17,7 @@ require (
 	github.com/onsi/gomega v1.16.0
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/pflag v1.0.5
+	github.com/vincent-petithory/dataurl v0.0.0-20191104211930-d1553a71de50
```

Comment:
This library is used by https://github.com/coreos/ignition to **encode** file contents into the ["data" URL scheme](https://datatracker.ietf.org/doc/html/rfc2397). Since we need to do the opposite process in our implementation here, I think it makes sense to use the same library for **decoding**. Furthermore, this operation is much more than a simple string manipulation and I couldn't figure out a way to achieve it using the standard library without re-implementing `github.com/vincent-petithory/dataurl`. Therefore, I chose to import the library.

I agree it isn't ideal that there are no official releases (or even tags) in that repo. If you have a suggestion for doing things differently @fabriziopandini, I'd be happy to hear.

Reply:
In slight defense, the repo is being packaged in Linux distributions in support of Ignition.
E.g., Fedora:

![image](https://user-images.githubusercontent.com/1441070/139849478-c7b2fe0d-3ce2-45e3-a724-ac12c5c03827.png)

Is this something Red Hat folk can work to resolve long term @enxebre ? Although, it won't help in this particular case since this is still going to be using the obsolete version of the Ignition repo.

Reply:
I won't block on this, but IMO this is something far from ideal, mostly because it seems we are being into an old, not maintained anymore, version of ignition without a path forward for upgrading sometimes down the line.
At least, let's discuss this at CAPI office hours

Reply:
This may have gotten lost in the complexity of the discussion, but we proposed just that: https://github.com/kubernetes-sigs/cluster-api/pull/4172/files#r740219714 - we have a path forward which we'd like to align on with you in the Nov 3rd office hours.
Would it make sense to separate concerns and discuss the dataurl import separately from the ignition one?

Reply:
Summing up our discussion from the Nov 3rd office hours:
- Reach out to the component's owner to discuss adding a release tag: (https://github.com/vincent-petithory/dataurl/issues/7)
-  We also offer help with maintenance to ensure this component continues to be supported.

Reply:
One alternative could be to Ignition module to offer some helper, as @fabriziopandini suggested to decode this, which seem useful to me. At least then it makes it an indirect dependency rather than a direct one. We would have to agree on it with upstream Ignition though to ensure forward-compatibility.

Reply:
v1 just got tagged BTW: https://github.com/vincent-petithory/dataurl/releases/tag/v1.0.0

Reply:
Cool.  Upstream Ignition update: https://github.com/coreos/ignition/pull/1280

Reply:
Pinned at `v1.0.0`.